src/utrfx/variant_util.py

```diff
@@ -24,50 +24,88 @@ def prepare_alt_seq(
         gene_chrom = region.contig
         gene_strand = region.strand
 
-    assert variant.region.contig == gene_chrom, "Variant and 5'UTR regions not in the same contig."
+    if variant.region.contig != gene_chrom: 
+        raise MismatchedContigsError()
 
     in_variant = any(region.overlaps_with(variant.region) for region in five_utrs.regions)
-    assert in_variant is True, "Variant not in the 5'UTR of the given transcript."
-
-    if variant.region.strand == gene_strand:
-        ref = variant.ref
-        alt = variant.alt
-        variant_region_start = variant.start
+    if in_variant is not True:
+        raise VariantNotIn5UTRError()
```

Comment:
This can be a bit subjective, but I would say, that checking if variant is in fact not in the `five_utrs` is not the responsibility of this method and `MismatchedContigs` and `VariantNotIn5UTRError` should not be raised. The same applies for `ReferenceMismatchError`.

If the variant is unrelated to `five_utrs`, then there is no work to be done and you can simply return the `cdna` sequence.

Checking for a reference base mismatch also sounds unrelated to the single responsibility of this function, and there should be a separate Q/C function for that.

So, in other words, the checks should probably be done, just not here. This will keep the `prepare_alt_seq` focused on its "thing".

However, if you think there are good reasons to check here, feel free to leave the function "as-is".